client/src/components/Navbar.jsx

```diff
@@ -409,12 +434,23 @@ const DesktopSubNav = ({ label, href, subLabel }) => {
                     });
                   };
 
+                  const matchCancelCallback = () => {
+                    modalComponentStore.setClosable(true);
+                    toast({
+                      title: `Cancelled match successfully`,
+                      status: "warning",
+                      duration: 5000,
+                      isClosable: true,
+                    });
+                  };
+
                   matchingFormStore
                     .startLoading()
                     .then(null, matchFailureCallback);
                   matchingFormStore.sendMatchRequest(
                     matchSuccessCallback,
-                    matchFailureCallback
+                    matchFailureCallback,
+                    matchCancelCallback
```

Comment:
I've merged the latest changes from the collab-svc branch.
You'll now need to pass `matchCancelCallback` to the modal for non-admin users too (because of changes to the Navbar's handling of admins and non-admins)

Reply:
As well has add the logic to disable the modal from closing while matching is in progess (ie force them to click "cancel")